lib/compiler/src/beam_utils.erl

```diff
@@ -197,10 +202,20 @@ bif_to_test('>', [A,B], Fail) -> {test,is_lt,Fail,[B,A]};
 bif_to_test('<', [_,_]=Ops, Fail) -> {test,is_lt,Fail,Ops};
 bif_to_test('>=', [_,_]=Ops, Fail) -> {test,is_ge,Fail,Ops};
 bif_to_test('==', [A,nil], Fail) -> {test,is_nil,Fail,[A]};
+bif_to_test('==', [nil,A], Fail) -> {test,is_nil,Fail,[A]};
+bif_to_test('==', [C,A], Fail) when ?is_const(C) ->
+    {test,is_eq,Fail,[A,C]};
```

Comment:
In arguments `[C,A]` while in return `[A,C]` - not a typo? Below `Ops` returns as is.

Reply:
Yes, not a typo, this is the entire point of the change.